Docs only:  format Command Line Options using a table

bradym opened the pull request:
## Description

This is a documentation only change, instead of wrapping the list of command line options in backticks, it puts them in a table. I made some minor updates to content for clarity and to fix some spelling errors.

## Motivation and Context

It's easier to read in table format, and it also eliminates the need to scroll horizontally to see the full description.

## How Has This Been Tested?

I've enabled github pages on my fork and you can see the result of this change here:
http://www.bradym.net/oauth2_proxy/docs/configuration.html

Once this has been merged, I'll disable the github pages setting on my fork.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

syscll commented:
**I really like this!** It makes the list of flags much more readable.

My only suggestions would be to wrap the flag/option name in single quotes and add a separate column for the type, maybe?

| Option | Type | Description |                                                                                                                                                        
| - | - | - |
| `-acr-values` | string | optional, used by login.gov (default "<http://idmanagement.gov/ns/assurance/loa/1">) |

Also, in a markdown table, the width of the columns don't have to be the exact width of the largest column. This makes it a lot easier to read the actual markdown as you don't have unnecessary space.  For example, the above table was created using the following MD:

```
| Option | Type | Description |                                                                                                                                                        
| - | - | - |
| `-acr-values` | string | optional, used by login.gov (default "<http://idmanagement.gov/ns/assurance/loa/1">) |
```

bradym commented:
I like your suggestion about having the type in a column, so I did that and added an additional column for default values as well. I also fixed a few spelling errors, added links to the text about "see the paragraph below" and reworded a few things to make them more clear.

A number of the options did not have types, so I added those. I also added the default values for flags to that enable behavior.

I saw that #165 is a docs formatting issue, so I fixed that while I'm at it since it was a very easy fix.

Oh, and I updated the description of this PR since I have now changed some content. :)

steakunderscore commented on docs/configuration/configuration.md:
should this: `| ------ | ---- | ----------- |`
be this?: `| ------ | ---- | ----------- | ------- |`

bradym replied:
Good catch, fixed!

JoelSpeed commented on docs/configuration/configuration.md:
```diff
@@ -134,7 +133,7 @@ For example, the `--cookie-secret` flag becomes `OAUTH2_PROXY_COOKIE_SECRET` and
 
 
-## Logging Configuration
+## <a anchor="logging-configuration"></a>Logging Configuration
```
I don't think this is necessary to set explicitly, it already works out of the box no?https://pusher.github.io/oauth2_proxy/docs/configuration#logging-configuration

bradym replied:
It looks like you're right. I saw that there was an <a> tag on the "Configuring for use with the Nginx `auth_request` directive" heading and my intention was to be consistent with what was being don in the doc already. Looks like my tag wasn't even setup right, should have been `< a name` instead of `<a anchor`.

I've removed that one and the one on the "Configuring for use with the Nginx `auth_request` directive" heading and confirmed that jekyll still renders the html correctly.

bradym replied:
I just noticed that the `<a>` tag next to the "Configuring for use with the Nginx auth_request directive" header was to provide a shorter link target for that header, so I've pushed another change to put that one back. Oops.
